### ISSUE TITLE
[Feature]

### ISSUE THREAD

**Issue by smulle48** [itavero/homebridge-z2m#479]
Please use the template and fill in all the data requested below!

**Problem description**
Since exposing groups by default, people using IKEA's old firmware for buttons are experiencing all bulbs, blinds, etc exposed twice.

In the latest firmware for IKEA buttons, they have removed group cast. The old way of controlling lights, was to add bulbs to an group and then bind the button to the group. That way you avoided the infamous "popcorn effect".

The downside of this nice feature, was that even if you would control just one bulb, it has to be in an group. This way you end up with the bulb itself and also an group for that same one bulb.

It could be fixed just by updating the firmware, but that would kill this nice feature the firmware have.

The good thing about the old firmware is that the group cast don't need the controller to be online to function, so even if the controller crash, you can still control all your light, blinds, etc.

**Suggested solution**
Disable exposing groups by default, and be able to expose them by their group id instead of the friendly name (friendly name accept spaces and non English letters)

**Describe alternatives you've considered**
Downgrading to an release prior to group exposure was default.

**Additional context**

**Are you willing to contribute to this feature?**
Yes

**Comment by itavero** [comment]
With regards to "disable exposing groups by default": I think for this, the easiest would be to add an additional configuration item where you can put in all the IDs or friendly names of the groups you do wish to expose.
If it is not in your configuration, all groups are exposed. If it is an empty array, none of the groups will be exposed.

Can you elaborate on "be able to expose them by their group id instead of the friendly name (friendly name accept spaces and non English letters)"?
You should be able to use both in the configuration already.

**Comment by smulle48** [comment]
> With regards to "disable exposing groups by default": I think for this, the easiest would be to add an additional configuration item where you can put in all the IDs or friendly names of the groups you do wish to expose.
> 
> If it is not in your configuration, all groups are exposed. If it is an empty array, none of the groups will be exposed.
> 
> 
> 
> Can you elaborate on "be able to expose them by their group id instead of the friendly name (friendly name accept spaces and non English letters)"?
> 
> You should be able to use both in the configuration already.

That sounds like an nice way to make it work.

The idea of using the id instead of the friendly name is, 1: you avoid troubles with misspelling and 2: if you rename the group, you don't hate to update the config, as long as the group id is the same.

**Comment by itavero** [comment]
> The idea of using the id instead of the friendly name is, 1: you avoid troubles with misspelling and 2: if you rename the group, you don't have to update the config, as long as the group id is the same.

But you can already use the group ID in the plugin configuration, so no change needed with regard to this?

**Comment by smulle48** [comment]
> > The idea of using the id instead of the friendly name is, 1: you avoid troubles with misspelling and 2: if you rename the group, you don't have to update the config, as long as the group id is the same.
> 
> 
> 
> But you can already use the group ID in the plugin configuration, so no change needed with regard to this?

I was only thinking about the groups  if it could support both it would be nice too 😁 

Just thought it would be easier to maintain on a daily basis, if one used the group id. So if I one day would like to expose group 17, I just put in that id, and not "Basement Gaming Room Second Row Spot" 😀 😀

**Comment by itavero** [comment]
> Just thought it would be easier to maintain on a daily basis, if one used the group id. So if I one day would like to expose group 17, I just put in that id, and not "Basement Gaming Room Second Row Spot" 😀 😀 

I agree. That's why for the current configuration options, you can already use both.

I think I'll be able to look at this request in more detail in the coming weeks. I'll keep you posted.

**Comment by jdtsmith** [comment]
Was the functionality _not_ to export groups ever introduced?  I use a groups to bind a remote to a bank of blinds, but I don't need/want them exposed to HomeKit.  Thanks for a great plugin.

**Comment by itavero** [comment]
If I recall correctly, I never made a single switch to exclude all groups in one go.
Individual groups can still be excluded by adding a configuration item to the `devices` list:

```jsonc
{
   "platform": "zigbee2mqtt",
   /* ... */
   "devices": [
      {
         "id": "some_group",
         "exclude": false
      },
      {
         "id": "some_other_group",
         "exclude": false
      }
   ]
}
```

**Comment by jdtsmith** [comment]
Thanks, that worked (with `"exclude": true`).
